### ISSUE TITLE
[Bug] ard_num_dims with IndependentMultitaskVariationalStrategy

### ISSUE THREAD

**Issue by valentinecesbio** [cornellius-gp/gpytorch#1659]
# 🐛 Bug

Hello, 

I am working on a multi-class (3 classes) classification using the dataset from sklearn `make_blobs` with 500 samples per class, and 20 features.
I would like to use the `ard_num_dims` option in order to have a separate lengthscale for each input dimension. I already use the option `batch_shape` because I want to learn a different set of variational parameters and hyperparameters for each output dimension.
As I want independent output dimensions, I do not want to use `LMCVariationalStrategy` but instead `IndependentMultitaskVariationalStrategy`. 

I got a correct classification with  `LMCVariationalStrategy` but it is not working with `IndependentMultitaskVariationalStrategy`.

Thanks for your help

## To reproduce
**Example with `LMCVariationalStrategy`**:

```python
import math
import torch
import gpytorch
import numpy as np
import tqdm.notebook
from matplotlib import pyplot as plt
from sklearn.datasets import make_blobs
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix as CM, accuracy_score as AS

print("numpy version: " + np.__version__)
print("gpytorch version: " + gpytorch.__version__)
print("torch version: " + torch.__version__)

nb_samples = 500
nb_features = 20
nb_classes = 3
random_state = 1
train_size = 0.75
number_inducing_points = 50

X_, y_ = make_blobs(n_samples=nb_samples, n_features=nb_features, centers=nb_classes, random_state=random_state)

X_train, X_val, y_train, y_val = train_test_split(X_, y_, train_size=train_size, random_state=random_state)

X_train = torch.tensor(X_train, dtype=torch.float)
y_train = torch.tensor(y_train , dtype=torch.float)
X_val = torch.tensor(X_val, dtype=torch.float)
y_val = torch.tensor(y_val, dtype=torch.float)

# Extract some random inducing points
torch.manual_seed(random_state)
# Returns a random permutation of integers from 0 to n - 1.
perm = torch.randperm(X_train.size(0))
idx = perm[:number_inducing_points]
# Same inducing points for each GP (size : nb_points_induits * nb_features)
inducing_points = X_train[idx, :]
# Different inducing points for each GP (size : nb_labels
# * nb_points_induits * nb_features)
inducing_points = np.repeat(inducing_points[np.newaxis, :, :], nb_classes, axis=0)

class GP_MultiClass_LMC(gpytorch.models.ApproximateGP):

    def __init__(self, inducing_points, nb_classes):

        # Define a Variational distribution
        variational_distribution = gpytorch.variational\
            .CholeskyVariationalDistribution(inducing_points.size(-2),
                                             batch_shape=torch.Size([nb_classes]))

        # MultitaskVariationalStrategy is deprecated, use
        # IndependentMultitaskVariationalStrategy instead.
        variational_strategy = gpytorch.variational.LMCVariationalStrategy(
            gpytorch.variational.VariationalStrategy(
                self, inducing_points, variational_distribution, learn_inducing_locations=True
            ),
            num_tasks=nb_classes,
            num_latents=nb_classes,
            latent_dim=-1,
        )

        super(GP_MultiClass_LMC, self).__init__(variational_strategy)
        self.mean_module = gpytorch.means.ConstantMean(
            batch_shape=torch.Size([nb_classes]))

        self.covar_module = gpytorch.kernels.ScaleKernel(
            gpytorch.kernels.RBFKernel(ard_nums_dim=nb_features,
                batch_shape=torch.Size([nb_classes])),
            batch_shape=torch.Size([nb_classes]), ard_num_dims=None)

         # Define a MultivariateNormal Distribution
    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        latent_pred = gpytorch.distributions.MultivariateNormal(mean_x,
                                                                covar_x)
        return latent_pred

# Define GP model
modelMulti = GP_MultiClass_LMC(inducing_points, nb_classes)

# Define a likelihood
likelihoodSoftmax = gpytorch.likelihoods.SoftmaxLikelihood(num_features=nb_classes, num_classes=nb_classes, mixing_weights=True)

num_epochs=100
learning_rate=0.1

# Find optimal model hyperparameters
modelMulti.train()
likelihoodSoftmax.train()

# Use the adam optimizer
optimizer = torch.optim.Adam([
    {'params': modelMulti.parameters()},
    {'params': likelihoodSoftmax.parameters()},
], lr=learning_rate)

mll = gpytorch.mlls.VariationalELBO(likelihoodSoftmax, modelMulti, y_train.numel())


epochs_iter = tqdm.notebook.tqdm(range(num_epochs), desc="Epoch")


for i in epochs_iter:
    # Within each iteration, we will go over each minibatch of data
    optimizer.zero_grad()
    output = modelMulti(X_train)
    loss = -mll(output, y_train)
    epochs_iter.set_postfix(loss=loss.item())
    loss.backward()
    optimizer.step()

# Get into evaluation (predictive posterior) mode
modelMulti.eval()
likelihoodSoftmax.eval()

# Test points are regularly spaced along [0,1]
# Make predictions by feeding model through likelihood
with torch.no_grad(), gpytorch.settings.fast_pred_var():
        observed_pred = likelihoodSoftmax(modelMulti(X_val))
        y_pred = observed_pred.probs.mean(0).argmax(-1)

conf_mat = CM(y_val, y_pred)
print(f"Confusion matrix:\n {conf_mat}")

accuracy = AS(y_val, y_pred)
print(f"Global accuracy: {accuracy}")

print(modelMulti.covar_module.base_kernel.lengthscale.shape)
print(modelMulti.covar_module.base_kernel.raw_lengthscale)
print(modelMulti.covar_module.base_kernel.raw_lengthscale_constraint)
```

**Output**
```
Confusion matrix:
 [[48  0  0]
 [ 0 35  0]
 [ 0  0 42]]
Global accuracy: 1.0
torch.Size([3, 1, 1])
Parameter containing:
tensor([[[-0.8027]],

        [[ 7.4186]],

        [[ 5.0510]]], requires_grad=True)
Positive()
```

**Example with `IndependentMultitaskVariationalStrategy`**:
```python
import math
import torch
import gpytorch
import numpy as np
import tqdm.notebook
from matplotlib import pyplot as plt
from sklearn.datasets import make_blobs
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix as CM, accuracy_score as AS

print("numpy version: " + np.__version__)
print("gpytorch version: " + gpytorch.__version__)
print("torch version: " + torch.__version__)

nb_samples = 500
nb_features = 20
nb_classes = 3
random_state = 1
train_size = 0.75
number_inducing_points = 50

X_, y_ = make_blobs(n_samples=nb_samples, n_features=nb_features, centers=nb_classes, random_state=random_state)

X_train, X_val, y_train, y_val = train_test_split(X_, y_, train_size=train_size, random_state=random_state)

X_train = torch.tensor(X_train, dtype=torch.float)
y_train = torch.tensor(y_train , dtype=torch.float)
X_val = torch.tensor(X_val, dtype=torch.float)
y_val = torch.tensor(y_val, dtype=torch.float)

# Extract some random inducing points
torch.manual_seed(random_state)
# Returns a random permutation of integers from 0 to n - 1.
perm = torch.randperm(X_train.size(0))
idx = perm[:number_inducing_points]
# Same inducing points for each GP (size : nb_points_induits * nb_features)
inducing_points = X_train[idx, :]
# Different inducing points for each GP (size : nb_labels
# * nb_points_induits * nb_features)
inducing_points = np.repeat(inducing_points[np.newaxis, :, :], nb_classes, axis=0)

class GP_MultiClass_Independent(gpytorch.models.ApproximateGP):

    def __init__(self, inducing_points, nb_classes):

        # Define a Variational distribution
        variational_distribution = gpytorch.variational\
            .CholeskyVariationalDistribution(inducing_points.size(-2),
                                             batch_shape=torch.Size([nb_classes]))

        # MultitaskVariationalStrategy is deprecated, use
        # IndependentMultitaskVariationalStrategy instead.
        variational_strategy = gpytorch.variational.IndependentMultitaskVariationalStrategy(
            gpytorch.variational.VariationalStrategy(
                self, inducing_points, variational_distribution,
                learn_inducing_locations=True
            ),
            num_tasks=nb_classes,
            )

        super(GP_MultiClass_Independent, self).__init__(variational_strategy)
        self.mean_module = gpytorch.means.ConstantMean(
            batch_shape=torch.Size([nb_classes]))

        self.covar_module = gpytorch.kernels.ScaleKernel(
            gpytorch.kernels.RBFKernel(ard_nums_dim=nb_features,
                batch_shape=torch.Size([nb_classes])),
            batch_shape=torch.Size([nb_classes]), ard_num_dims=None)

         # Define a MultivariateNormal Distribution
    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        latent_pred = gpytorch.distributions.MultivariateNormal(mean_x,
                                                                covar_x)
        return latent_pred

# Define GP model
modelMulti = GP_MultiClass_Independent(inducing_points, nb_classes)

# Define a likelihood
likelihoodSoftmax = gpytorch.likelihoods.SoftmaxLikelihood(num_features=nb_classes, num_classes=nb_classes, mixing_weights=True)

num_epochs=100
learning_rate=0.1

# Find optimal model hyperparameters
modelMulti.train()
likelihoodSoftmax.train()

# Use the adam optimizer
optimizer = torch.optim.Adam([
    {'params': modelMulti.parameters()},
    {'params': likelihoodSoftmax.parameters()},
], lr=learning_rate)

mll = gpytorch.mlls.VariationalELBO(likelihoodSoftmax, modelMulti, y_train.numel())


epochs_iter = tqdm.notebook.tqdm(range(num_epochs), desc="Epoch")


for i in epochs_iter:
    # Within each iteration, we will go over each minibatch of data
    optimizer.zero_grad()
    output = modelMulti(X_train)
    loss = -mll(output, y_train)
    epochs_iter.set_postfix(loss=loss.item())
    loss.backward()
    optimizer.step()

# Get into evaluation (predictive posterior) mode
modelMulti.eval()
likelihoodSoftmax.eval()

# Test points are regularly spaced along [0,1]
# Make predictions by feeding model through likelihood
with torch.no_grad(), gpytorch.settings.fast_pred_var():
        observed_pred = likelihoodSoftmax(modelMulti(X_val))
        y_pred = observed_pred.probs.mean(0).argmax(-1)
```
**Output**
```
Confusion matrix:
 [[ 0 48  0]
 [ 0 35  0]
 [ 0 42  0]]
Global accuracy: 0.28
torch.Size([3, 1, 1])
Parameter containing:
tensor([[[-1.2953]],

        [[-0.6173]],

        [[ 0.0733]]], requires_grad=True)
Positive()
```
## Expected Behavior

**First question:** 

Why my lengthscale is not a `c x 1 x d` tensor but a `c x 1 x 1` tensor with `batch_shape=nb_classes=c` and `ard_num_dims=d`  ?

**Second question:**

Why the optimization is well working with `LMCVariationalStrategy` but it is not working with `IndependentMultitaskVariationalStrategy` when I add the option `ard_num_dims` ?

## System information

**Please complete the following information:**
gpytorch version: 1.2.0
torch version: 1.3.1
Ubuntu 20.04.1 LTS

## Additional context
Maybe this is related to #1209

**Comment by gpleiss** [comment]
> Why my lengthscale is not a c x 1 x d tensor but a c x 1 x 1 tensor with batch_shape=nb_classes=c and ard_num_dims=d ?

There is a typo in your code. The argument should be `ard_num_dims`, not `ard_nums_dim`.

> Why the optimization is well working with LMCVariationalStrategy but it is not working with IndependentMultitaskVariationalStrategy when I add the option ard_num_dims ?

1. You should set `mixing_weights=False` in both of your examples. They are redundant in the LMC case, and they make your GPs dependent in the `IndependentMultitaskVariationalStrategy` case. Removing them solves the issue
2. If you make the learning rate smaller for the likelihood parameters, that also fixes the issue. But again, I'd just remove the `mixing_weight` parameters.

P.S. The `mixing_weights` feature is actually going to be deprecated, since it replicates the functionality of the LMC variational strategy and it is less stable

**Comment by valentinecesbio** [comment]
Thank you very much for your help!
